lib/route-handler.js

```diff
@@ -85,7 +95,12 @@ export default class RouteHandler {
         this.handler.setRequest(request);
       }
 
-      result = this.handler.handle(request);
+      result = this.handleWithMiddleware(request, [
+        ...middleware,
+        () => {
+          return this.handler.handle(request);
```

Comment:
With this approach, if middleware wants to change the request, it needs to mutate the `request` parameter directly before calling `next()`, right?  What would you think about also allowing middleware to call `next(request)`, to be able to pass a cloned or even completely new request object into the subsequent handler?  I think something like this could work, while still allowing `next()` to be called without arguments, right?

```suggestion
        (req = request) => {
          return this.handler.handle(req);
```

Reply:
I like the idea quite a bit; it'll need a couple more lines of code and type definitions, too - I'll get on that.  I haven't made new `Requests` from old `Requests`, but as long as all the data is available to copy over it sounds sound.

Removing `withMiddleware` seems like a good idea, too.  I think it's easy enough for library consumers to write a standalone version of that if they need to once the simpler API is available.  We can always add it at some point if there's demand.

Reply:
Updated :-)